xbmc/addons/AddonManager.cpp

```diff
@@ -42,6 +43,7 @@ namespace ADDON
  */
 
 std::map<TYPE, IAddonMgrCallback*> CAddonMgr::m_managers;
+CAddonRepos RepoInfos;
```

Comment:
Global variables is very bad design, as you do not have control when the instance will be destroyed. Make it a member of CAddonManager.

Reply:
cannot use a member (or don't know how to use it) because then i'm getting trouble with const correctness. but there's a new version going up shortly that uses local instantiation instead of global.

when trying to use a private member i get:
```
/home/fhw/xbmc/xbmc/addons/AddonManager.cpp: In member function ‘ADDON::VECADDONS ADDON::CAddonMgr::GetAvailableUpdates() const’:
/home/fhw/xbmc/xbmc/addons/AddonManager.cpp:222:50: error: passing ‘const ADDON::CAddonRepos’ as ‘this’ argument discards qualifiers [-fpermissive]
   m_repoInfos.SetupFullAddonsByRepoMap(all_addons);
                                                  ^

```

Reply:
Don’t worry this too much at the moment. Let’s continue on the draft a bit and we add/remove this later.

I think we can probably get away with declaring this locally where it’s needed in the end.